packages/visual-editor/src/hooks/useEditorModeClassName.ts

```diff
@@ -23,22 +23,35 @@ export const useEditorModeClassName = ({
       return;
     }
 
-    const [className, styleRule] = buildStyleTag({ styles, nodeId });
+    const [newClassName, styleRules] = buildStyleTag({ styles, nodeId });
+    addStylesTag(newClassName, styleRules);
 
-    setClassName(className);
+    if (className !== newClassName) {
+      setClassName(newClassName);
+      // Clean up: remove outdated styles from DOM
+      removeStylesTag(className);
+    }
+  }, [styles, nodeId, className]);
 
-    const existingTag = document.querySelector(`[data-cf-styles="${className}"]`);
+  return className;
+};
 
-    if (existingTag) {
-      existingTag.innerHTML = styleRule;
-      return;
-    }
+const removeStylesTag = (className: string) => {
+  const existingTag = document.querySelector(`[data-cf-styles="${className}"]`);
+  if (existingTag) {
+    document.head.removeChild(existingTag);
+  }
+};
```

Comment:
I added some local debug logging to this method and never observed it finding an existing tag to remove. I could only see this function getting called with an empty string for the className (from the initial state definition).

Do you know of a scenario that would trigger this function to remove an existing style tag from the dom? Is there a test that could be written to verify the behavior?

Reply:
You're right that this scenario currently is not occurring as every node has an id. I pointed that out in the PR description as well because I only noticed it late but continued to make the function "saver".